#This file merge burn severity and recovery data for defoliated fires.

In [146]:
import plotly.express as px
import pandas as pd
import statsmodels
import numpy as np
from pathlib import Path


Read in files

In [147]:
dfrbr = pd.read_csv("/Users/jgoldman/Library/CloudStorage/OneDrive-UniversityofToronto/Data/fire-only-sbw-only-perimeters/cleaned/rbr/defol/defoliated-perims-rbr-v2.csv")
dfnbr = pd.read_csv("/Users/jgoldman/Library/CloudStorage/OneDrive-UniversityofToronto/Data/fire-only-sbw-only-perimeters/cleaned/nbr-recovery/defol/defoliated_fires_nbr.csv")


What is the length of each dataset?

In [148]:
len(dfrbr)

321

In [149]:
len(dfnbr)

312

There are more rbr values 321 than there are nbr values 312, check which ones are missing in nbr

In [150]:
fireID_rbr = dfrbr['Fire_ID'].to_list()
fireID_nbr = dfnbr['Fire_ID'].to_list()
list(set(fireID_rbr) - set(fireID_nbr))

['NOR56_2018_2044',
 'NOR69_2018_2119',
 'SUD35_2016_2006',
 'NOR55_2018_2141',
 'NOR68_2018_2028',
 'NOR62_2018_2053',
 'PAR33_2018_2071',
 'NOR25_2018_2121',
 'NOR42_2018_2129']

Remove fires that dont match in df_rbr

In [151]:
remove_ids = list(set(fireID_rbr) - set(fireID_nbr))

print(remove_ids)


['NOR56_2018_2044', 'NOR69_2018_2119', 'SUD35_2016_2006', 'NOR55_2018_2141', 'NOR68_2018_2028', 'NOR62_2018_2053', 'PAR33_2018_2071', 'NOR25_2018_2121', 'NOR42_2018_2129']


In [152]:
dfrbr = dfrbr[~dfrbr['Fire_ID'].isin(remove_ids)]
len(dfrbr)
dft = dfrbr[(dfrbr['Fire_ID']) == 'NOR56_2018_2044']

I kept the defol column, this defol column is has a few zeros, check the data to see the size of these perimeters.

In [153]:
dfrbr = dfrbr[['Fire_ID', 'rbrCV', 'rbrExtreme', 'rbrMedian', 'tsd', 'defol', 'years_defol']]
len(dfrbr)

312

In [154]:
dfnbr = dfnbr[['Fire_ID', 'nbr1', 'nbr2','nbr3', 'nbr4', 'nbr5', 'nbr6', 'nbr7', 'nbr8', 'nbr9', 'nbr10']]
dfnbr.head()


Fire_ID      nbr1      nbr2      nbr3      nbr4      nbr5  \
0   CHA12_1991_532  0.205961  0.204704  0.279267  0.233863  0.204076   
1   GOG4_1991_1464  0.188382  0.249941  0.230050  0.194282  0.218880   
2   GER40_1991_716  0.131757  0.155174  0.201234  0.186374  0.158652   
3   HEA21_1991_512  0.106197  0.167786  0.228341  0.165205  0.144869   
4  CHA52_1991_1021  0.142085  0.210453  0.201205  0.191209  0.201068   

       nbr6      nbr7      nbr8      nbr9     nbr10  
0  0.251705  0.269388  0.287662  0.235703  0.249931  
1  0.233660  0.264350  0.286031  0.244063  0.240820  
2  0.204622  0.164004  0.211810  0.207940  0.250839  
3  0.200430  0.195326  0.208556  0.194975  0.253994  
4  0.203699  0.255377  0.273232  0.234927  0.262036

In [155]:
fireID_rbr = dfrbr['Fire_ID'].to_list()
fireID_nbr = dfnbr['Fire_ID'].to_list()
list(set(fireID_rbr) - set(fireID_nbr))

[]

In [156]:
df_defol = dfnbr.merge(dfrbr, on='Fire_ID')

len(df_defol)

324

There seems to be duplicated columns. check the duplicates

In [157]:
df_dup = df_defol[df_defol.duplicated()]

In 1992, there are 6 triplicated fires. When checking the based csv files, it seems like the perimeters are in the defoliated area and not the non-defoliated area. This may be either completely defoliated fires, or areas whether the split did not occur properly. This needs to be investigated further. For now, these fires will be removed and I will continue with only 306 fires.

In [158]:
# remove duplicate fires
df_defol_v2 = df_defol.drop_duplicates()



In [159]:
outfolder = Path("/Users/jgoldman/Library/CloudStorage/OneDrive-UniversityofToronto/Data/fire-only-sbw-only-perimeters/cleaned/merged/")
df_defol_v2.to_csv(outfolder / "defoliated_merged.csv")
df_dup.to_csv(outfolder / "defoliated_merged_removed_duplicates.csv")